File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def solution(video_len, pos, op_start, op_end, commands):
    video_split = list(map(int, video_len.split(":")))
    video_time = video_split[0] * 60 + video_split[1];
    
    op_split = [list(map(int, op_start.split(":"))), list(map(int, op_end.split(":")))]
    op_start_time = op_split[0][0] * 60 + op_split[0][1]
    op_end_time = op_split[1][0] * 60 + op_split[1][1]
    
    pos_split = list(map(int, pos.split(":")))
    pos_time = pos_split[0] * 60 + pos_split[1];
    
    if op_start_time <= pos_time <= op_end_time:
        pos_time = op_end_time
    
    for command in commands:
        if command == 'prev':
            pos_time -= 10
            if pos_time < 0:
                pos_time = 0
        if command == 'next':
            pos_time += 10
            if pos_time > video_time:
                pos_time = video_time
        if op_start_time <= pos_time <= op_end_time:
            pos_time = op_end_time

    div, mod = map(str, divmod(pos_time, 60))
    return div.zfill(2) + ":" + mod.zfill(2)

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.java
class Solution {
    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        String[] videoSplit = video_len.split(":");
        int videoTime = Integer.parseInt(videoSplit[0]) * 60 + Integer.parseInt(videoSplit[1]);
        
        String[] opStartSplit = op_start.split(":");
        int opStartTime = Integer.parseInt(opStartSplit[0]) * 60 + Integer.parseInt(opStartSplit[1]);
        String[] opEndSplit = op_end.split(":");
        int opEndTime = Integer.parseInt(opEndSplit[0]) * 60 + Integer.parseInt(opEndSplit[1]);
       
        String[] posSplit = pos.split(":");
        int posTime = Integer.parseInt(posSplit[0]) * 60 + Integer.parseInt(posSplit[1]);
        
        if (posTime >= opStartTime && posTime <= opEndTime) {
            posTime = opEndTime;
        }
        
        for (String command : commands) {
            if (command.equals("prev")) {
                posTime -= 10;
                if (posTime < 0) {
                    posTime = 0;
                }
            }
            if (command.equals("next")) {
                posTime += 10;
                if (posTime > videoTime) {
                    posTime = videoTime;
                }
            }
            if (posTime >= opStartTime && posTime <= opEndTime) {
                posTime = opEndTime;
            }
        }
        
        int div = (int) posTime / 60; int mod = posTime % 60;
        StringBuilder sb = new StringBuilder();
        sb.append(String.format("%02d", div)).append(":").append(String.format("%02d", mod));
        return sb.toString();
    }
}